pydantic/_internal/_generate_schema.py

```diff
@@ -450,13 +450,14 @@ def generate_dc_field_schema(
         schema = self.generate_schema(field_info.annotation)
         schema = apply_annotations(schema, field_info.metadata)
 
-        if not field_info.is_required():
-            schema = wrap_default(field_info, schema)
-
         schema = apply_validators(
             schema, filter_field_decorator_info_by_field(decorators.field_validator.values(), name)
         )
         schema = apply_validators(schema, filter_field_decorator_info_by_field(decorators.validator.values(), name))
+
+        if not field_info.is_required():
+            schema = wrap_default(field_info, schema)
+
```

Comment:
This is the same issue we've run into before where if you put a function validator outside of a WithDefault then the WithDefault will not work. This fixes it and doesn't seem to break anything, but there is also the option of making WithDefault recursive in pydantic-core (i.e. the function validator would call into the schema it wraps to see if it has a default). This would make sense for function validators, but I'm not sure where else, if it even makes sense in general or if there are even any cons to just forcing WithDefault to be the outermost schema.